### DISCUSSION TITLE
Unit Test

### DISCUSSION THREAD

**Discussion by Julian-B90** [stephanrauh/ngx-extended-pdf-viewer#2496] [Ideas]
Is it planned to create a unit test?

**Comment by stephanrauh** [comment]
Not really. The library is stable enough now that unit tests might pay off, but I don't have enough time to write them.

I was wondering whether a friendly AI could generate unit tests for me, but so far, the results were underwhelming.

**Reply by Julian-B90** [reply]
Maybe i can start it. For the template i use Copilot with some optimizations.

**Reply by stephanrauh** [reply]
Awesome, thanks in advance!

**Reply by stephanrauh** [reply]
Thing is, the unit tests would find only a few bugs. For example, it's unlikely someone writes a unit test detecting #2493. That's why I focus on the Playwright tests. But of course, at the moment even the Playwright tests are outdated... :)

Nonetheless, I'd appreciate it if you'd contribute tests. npm doesn't like projects without tests.

**Reply by Julian-B90** [reply]
What do you think about https://ngneat.github.io/spectator/ or do you want to stay with Jest only?

**Reply by stephanrauh** [reply]
You're the expert. If you've gained good experience with Spectator, go for it. My first impression it's like ngMocks on steroids.
